nilearn/connectome/connectivity_matrices.py

```diff
@@ -231,6 +231,77 @@ def sym_to_vec(symmetric, discard_diagonal=False):
     return symmetric[..., tril_mask] / scaling[tril_mask]
 
 
+def vec_to_sym_matrix(vec, diagonal=None):
+    """Return the symmetric matrix given its flattened lower triangular part.
+    Acts on the last dimension of the array if not 1-dimensional.
+    Diagonal can be encompassed in vec or given separately. In both cases, note
+    that diagonal elements are multiplied by sqrt(2).
+
+    Parameters
+    ----------
+    vec : numpy.ndarray, shape (..., p * (p + 1) /2)
+        The input array.
+
+    diagonal : numpy.ndarray, shape (..., p), optional
+        The diagonal array to be stacked to vec.
+
+    Returns
+    -------
+    sym : numpy.ndarray, shape (..., p, p) or (..., p + 1, p + 1) if diagonal
+        is given seperately.
+        The output symmetric array.
+
+    Note
+    ----
+    This function is meant to be the inverse of sym_matrix_to_vec.
+
+    See also
+    --------
+    nilearn.connectome.sym_matrix_to_vec
+    """
+    n = vec.shape[-1]
+    # solve p * (p + 1) / 2 = n subj. to p > 0
+    # p ** 2 + p - 2n = 0 & p > 0
+    # p = - 1 / 2 + sqrt( 1 + 8 * n) / 2
+    p = (sqrt(8 * n + 1) - 1.) / 2
+    if p > floor(p):
+        raise ValueError(
+            "Vector of unsuitable shape {0} can not be transformed to "
+            "a symmetric matrix.".format(vec.shape))
+
+    p = int(p)
+    first_shape = vec.shape[:-1]
+    if diagonal is None:
+        n_columns = p
+        diagonal_offset = 0
+    else:
+        if diagonal.shape[:-1] != first_shape or diagonal.shape[-1] != p + 1:
```

Comment:
>I didn't understand the condition here, `diagonal.shape[-1] != p + 1`

I have to correct the doc, but `p + 1` is correct here.
Your questions tell me my code is unclear and I have to rework it. May be I can add a little function that computes the shape of the symmetric matrix from the given vector, and also add a section example to illustrate how it works: if you enter a vector that does not include the diagonal in it, the computed shape from the equation is less than the true matrix shape
```Python
sym = np.array([[1., .3, .1], [.3, 1., .5], [.1, .5, 1.]])
vec = sym_matrix_to_vec(sym, discard_diagonal=True)
# Recover the matrix shape
n = np.shape(vec)[-1]
p = (sqrt(8 * n + 1) - 1.) / 2
n_columns = int(p) + 1
print n_columns
```
3
```Python
diagonal = np.ones(n_columns,) / sqrt(2)
sym2 = vec_to_sym_matrix(vec, diagonal)
print sym2
```
outs
```Python
[[ 1.   0.3  0.1]
 [ 0.3  1.   0.5]
 [ 0.1  0.5  1. ]]
```